homeassistant/components/humidifier/__init__.py

```diff
@@ -38,6 +38,9 @@
     DEVICE_CLASS_DEHUMIDIFIER,
     DEVICE_CLASS_HUMIDIFIER,
     DOMAIN,
+    MODE_AUTO,
```

Comment:
These do not seem to be used in the code base, so I didn't include them but I can add in if someone feels they should be.
```python
MODE_ECO = "eco"
MODE_BOOST = "boost"
MODE_COMFORT = "comfort"
MODE_HOME = "home"
MODE_SLEEP = "sleep"
MODE_BABY = "baby"
```

Reply:
This does look like a good candidate for StrEnum.

Reply:
Similar to `climate` entity, `xiaomi_miio` uses custom modes, so I don't think we should convert this to StrEnum
cc. @MartinHjelmare

Reply:
I have opened #78181 for initial feedback

Reply:
I have moved the logbook changes out, so it can be merged before humidifer discussion is resolved.
I'll mark this as draft in the meantime.